documentation/docs/extensibility.md

```diff
@@ -1,86 +1,184 @@
+
+```
+./vars/myCustomPipeline.groovy
+./README.md
+```
+
+where `myCustomPipeline.groovy` contains the modified pipeline code.
+
+!!! note
+    Your custom pipeline _must_ be named different from the other pipelines provided by project "Piper", because Jenkins requires names across multiple libraries to be unique.
+
+This library needs to be placed in a git repository which is available for Jenkins and must be configured in Jenkins [as documented here](https://jenkins.io/doc/book/pipeline/shared-libraries/#using-libraries).
+
+![Library Setup](images/customPipelineLib.png "Library Setup")
+
+The `Jenkinsfile` would look similar to this:
+
+```groovy
+@Library(['piper-lib-os@vINSERT_VERSION_HERE','my-own-pipeline@vINSERT_VERSION_HERE']) _
+
+myCustomPipeline script: this
+```
+
+Be sure to adapt the names and version identifiers accordingly.
+
+### How to stay up-to-date
+
+Regardless which of the above options you chose, one downside of this approach is that your pipeline will be out of sync with the centrally maintained pipelines at some point in time.
+We strongly recommend doing _as little modification as possible_ to fulfil your requirements.
+Please be aware that stages might have dependencies on each other.
+Your pipeline should treat _stages_ as a black box, the implementation of stages is no published API and may be subject to change at any time.
+
+!!! warning "Beware of breaking changes"
+    Please be aware that when using the `master` branch of a library, it might always happen that breaking changes occur.
+    We recommend to always fix versions to a released version like in this example: `@Library('my-shared-library@1.0') _`<br />
+    Find the most recent release for [jenkins-library](https://github.com/SAP/jenkins-library/releases) and for [SAP Cloud SDK Pipeline](https://github.com/SAP/cloud-s4-sdk-pipeline/releases) on GitHub.
+    We do recommend to ["watch" releases for those repositories on GitHub](https://help.github.com/en/github/receiving-notifications-about-activity-on-github/watching-and-unwatching-releases-for-a-repository).
+
+!!! note "When to go with a modified ready-made pipeline"
+    This option is right for you, when none of the provided ready-made pipelines serves your purpose, and individual stage extensions don't provide enough flexibility.
+
+### Advanced tips and information
+
+When you consider adding additional capabilities your first stop should be the [Jenkins Pipeline Steps Reference](https://jenkins.io/doc/pipeline/steps/).
+Here you get an overview about what kind of capabilities are already available, and a list of related parameters which you can use to customize the existing implementation.
+The provided information should help you to understand and extend the functionality of your pipeline.
+
+## 3. Custom `Jenkinsfile`
+
+Since project "Piper" fully builds on [Jenkins Pipelines as Code](https://jenkins.io/doc/book/pipeline-as-code/), you can also go with your complete custom `Jenkinsfile`.
```

Comment:
This may be irritating from the user's perspective: The "Jenkinsfile" is initially provided in the architype. When you modify it according to approach 2), then this would already be a considered a "complete custom Jenkinsfile". Maybe distiguish with "pipeline written from scratch"?